tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -217,22 +217,39 @@ public function adminHasDeletedGroupUsingTheGraphApi(string $group): void {
 	 * sends a request to delete a user using the Graph API
 	 *
 	 * @param string $user username is used as the id
+	 * @param string|null $userAsAdmin
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 */
-	public function adminDeletesUserUsingTheGraphApi(string $user): void {
+	public function adminDeletesUserUsingTheGraphApi(string $user, string $userAsAdmin = null): void {
+		$credentials = $this->getAdminOrUserCredentials($userAsAdmin);
+
 		$this->featureContext->setResponse(
 			GraphHelper::deleteUser(
 				$this->featureContext->getBaseUrl(),
 				$this->featureContext->getStepLineRef(),
-				$this->featureContext->getAdminUsername(),
-				$this->featureContext->getAdminPassword(),
+				$credentials["username"],
+				$credentials["password"],
 				$user
 			)
 		);
 	}
 
+	/**
+	 * @When /^the user "([^"]*)" deletes a user "([^"]*)" using the Graph API$/
+	 *
+	 * @param string $userAsAdmin
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function theUserDeletesAUserUsingTheGraphAPI(string $userAsAdmin, string $user): void {
+		$this->adminDeletesUserUsingTheGraphApi($user, $userAsAdmin);
+	}
```

Comment:
this wrapping will only add depth. can we manage the step in the fn `adminDeletesUserUsingTheGraphApi` func?

Reply:
Thanx :+1: 
I tried to manage the step in method `adminDeletesUserUsingTheGraphApi`, but I think we cannot do that, `adminDeletesUserUsingTheGraphApi` takes an optional parameter `$userAsAdmin`, the order of parameters we get from `When the user "Alice" deletes a user "nonExistentUser" using the Graph API` won't match with method `adminDeletesUserUsingTheGraphApi`.